riptable/rt_str.py

```diff
@@ -915,5 +874,81 @@ def char(self, position: Union[int, List[int], np.ndarray]):
     nb_char_par = _njit_par(_nb_char)
 
 
+MIN_INT = np.iinfo(np.int32).min
+
+
+def _populate_wrappers(cls):
+    """
+    Decorator for the CatString class which populates the string methods and sets the defaults for
+    filtered value fills.
+    """
+    for method, filter_value_fill in [
+        ('upper', None),
+        ('lower', None),
+        ('reverse', None),
+        ('removetrailing', None),
+        ('strlen', MIN_INT),
+        ('index_any_of', MIN_INT),
+        ('index', MIN_INT),
+        ('contains', False),
+        ('startswith', False),
+        ('endswith', False),
+        ('regex_match', False),
+        ('substr', None),
+        ('char', None)
+    ]:
+        setattr(cls, method, cls._build_method(method, filter_value_fill))
+    return cls
+
+
+@_populate_wrappers
+class CatString:
+    """
+    Provides access to FAString methods for Categoricals.
+    All string methods are wrappers of the FAString equivalent with
+    categorical re-expansion and option for how to fill filtered elements.
+    """
+    def __init__(self, cat):
+        from .rt_categorical import CategoryMode
+        self.cat = cat
+        if cat.isenum:

```

Comment:
For maximum robustness, you need to check the ``.category_mode`` property instead of ``.isenum``, ``.issinglekey``, .ismultikey`` -- ``.issinglekey`` covers both integer and string-based single-key categoricals. As-is, if someone has an integer-based ``Categorical`` and calls ``.str`` on it, it'll pass the ``.issinglekey`` check you have below. It should then fail in the ``FAString.__init__()`` but it'd be better if we fell through to the ``else`` case here since it provides a better error message to the user.

Reply:
Got it. I just copied this piece from Categorical.str.fget.